File: library/src/main/java/de/nenick/espressomacchiato/elements/EspEditText.java
package de.nenick.espressomacchiato.elements;

import android.support.test.espresso.action.ViewActions;
import android.view.View;

import org.hamcrest.Matcher;

import de.nenick.espressomacchiato.matchers.EspAllOfBuilder;

import static android.support.test.espresso.assertion.ViewAssertions.matches;
import static android.support.test.espresso.matcher.ViewMatchers.withHint;
import static android.support.test.espresso.matcher.ViewMatchers.withText;

public class EspEditText extends EspView {

    public static EspEditText byId(int resourceId) {
        return new EspEditText(resourceId);
    }

    public static EspAllOfBuilder<EspEditText> byAll() {
        return new EspAllOfBuilder<EspEditText>() {};
    }

    public EspEditText(int resourceId) {
        super(resourceId);
    }

    public EspEditText(Matcher<View> base) {
        super(base);
    }

    public void replaceText(String newText) {
       findView().perform(ViewActions.replaceText(newText));
    }

    public void assertTextIs(String expectedText) {
        findView().check(matches(withText(expectedText)));
    }

    public void assertHintTextIs(String expectedText) {
        findView().check(matches(withHint(expectedText)));
    }
}

File: esp-core/src/main/java/de/nenick/espressomacchiato/elements/basics/EspEditText.kt
package de.nenick.espressomacchiato.elements.basics

import android.view.View
import androidx.annotation.IdRes
import androidx.test.espresso.Root
import androidx.test.espresso.matcher.ViewMatchers
import de.nenick.espressomacchiato.espresso.view.TextActions
import de.nenick.espressomacchiato.espresso.view.TextAssertions
import de.nenick.espressomacchiato.internals.OpenForExtensions
import org.hamcrest.Matcher

@OpenForExtensions
class EspEditText(
        viewMatcher: Matcher<View>,
        rootMatchers: Matcher<Root>? = null,
        interactions: EspEditText.() -> Unit = {}
) :
        EspView(viewMatcher, rootMatchers, interactions as EspView.() -> Unit),
        TextActions,
        TextAssertions {

    constructor(@IdRes viewId: Int = 0, interactions: EspEditText.() -> Unit = {})
            : this(viewId, null, interactions)

    constructor(@IdRes viewId: Int = 0, rootMatchers: Matcher<Root>? = null, interactions: EspEditText.() -> Unit = {})
            : this(ViewMatchers.withId(viewId), rootMatchers, interactions)
}